docs/development-e2e-setup.md

```diff
@@ -0,0 +1,85 @@
+# Development end-to-end setup
+
+
+## Setup
+To run all the applications in a fashion that enables a provider to submit a claim or application and then see that in the case worker app for assessment you can follow this guide.
+
+For provider app submission you will need to ensure you have the following env vars in your `.env.development.local`. These will enable authenticated communication with the local running version of the app store. Ensure it is the `...local` version and that no secrets are inadvertently committed to the repository.
+
+```sh
+# App store credentials (for authenticating to a local app store currently)
+# TODO: Oauth may be removed from local/development app store in the future
+APP_STORE_CLIENT_ID=not-a-real-key
+APP_STORE_CLIENT_SECRET=not-a-real-key
+APP_STORE_TENANT_ID=not-a-real-key
+```
+Ask a dev where these can be retrieved, and do not use production versions
+
+To receive emails to you own email address, or avoid errors on submission, you should supply the following ENV VAR
+```sh
+GOVUK_NOTIFY_API_KEY=not-a-real-key
+```
+You can generate your own "test" key (that does not send emails) or a limited key (that only sends to a controlled list of emails). You will need access to https://www.notifications.service.gov.uk/ or our existing secrets to supply this value. Ask a dev.
+
+To login to your locally running sidekiq web user interface you will need to add these ENV vars. The values are just for your local Sidekiq webUI basic authentication. This webUI can be useful to see Job success, failure and errors.
+
+```sh
+SIDEKIQ_WEB_UI_USERNAME=sidekiq
+SIDEKIQ_WEB_UI_PASSWORD=sidekiq
+```
+
+## Running e2e development stack
+
+### Run the provider App:
+  - Follow above readme to setup the development requirements
+
+  - Amend Procfile.dev to run rails server on a unique port of your choice
```

Comment:
I'd almost recommend we hard-code 3001 into the procfile. If we're all going to have both caseworker and provider up and running simultaneously more and more, maybe we should all get into the habit of having one on 3001 and the other on 3002 or something, by having those as standard in the procfile.

Reply:
i did muck about with `${PORT:-3000}` type logic but there are so many ways it seems to go wrong (puma config, foreman start command, Procfile.dev) that i have never found a robust solution, here or elsewhere.

So i would vote for hardcoding 3001/3002 into provider and caseworker 👍

Reply:
Agreed with this